dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -609,11 +634,159 @@ static DockerSpecifier determineImageSpecifier(String image, DockerImageReferenc
+                // Check that the image manifest is using schema version 2 because schema version 1 is deprecated and the JSON response is
+                // formatted differently, thus requiring some pre-processing before calculating the image digest
+                if (imageManifest.getSchemaVersion() != 2) {
+                    LOG.error("The image manifest is using schema version {}, not schema version 2", imageManifest.getSchemaVersion());
+                    return Optional.empty();
+                }
+
+                // An image's digest is the sha256 of its image manifest. Source: https://docs.docker.com/registry/spec/api/#content-digests
+                digest = Hashing.sha256().hashString(imageManifestJson, StandardCharsets.UTF_8).toString();
+                checksum = new Checksum("sha256", digest);
+            } else {
+                digest = digestHeader.getValue();
+                checksum = new Checksum(digest.split(":")[0], digest.split(":")[1]);
+            }
+
+            List<Checksum> checksums = Collections.singletonList(checksum);
+
+            // An image's size is the sum of the size of its layers
+            List<DockerLayer> imageLayers = Arrays.asList(imageManifest.getLayers());
+            long imageSize = imageLayers.stream().mapToLong(DockerLayer::getSize).sum();
+
+            // imageTag is null if the image is specified by digest because the corresponding tag is not provided in the image manifest
+            String imageTag = null;
+            if (specifierType == DockerSpecifier.TAG) {
+                imageTag = specifierName;
+            }
+
+            // Download the blob for the config to get architecture and os information
+            String configDigest = imageManifest.getConfig().getDigest();
+            String getBlobURL = String.format(getBlobURLFormat, registry.getDockerPath(), repo, configDigest);
+            Optional<String> configBlobResponse = ResourceUtilities.asString(getBlobURL, token, client);
+
+            if (configBlobResponse.isEmpty()) {
+                LOG.error("Could not retrieve config blob for {} image {} specified by {} with value {}", registry.getFriendlyName(), repo,
+                        specifierName, specifierType);
+                return Optional.empty();
+            }
+
+            DockerBlob configBlob = GSON.fromJson(configBlobResponse.get(), DockerBlob.class);
+            String arch = configBlob.getArchitecture();
+            String os = configBlob.getOs();
+
+            // Note: Unable to get imageID and imageUpdateDate from the image's manifest
+            Image image = new Image(checksums, repo, imageTag, null, registry, imageSize, null);
+            image.setSpecifier(specifierType);
+            image.setArchitecture(arch);
+            image.setOs(os);
+
+            return Optional.of(image);
+        } finally {
+            httpRequest.releaseConnection();
```

Comment:
Might be more clear when this gets tested, but I think `releaseConnection` is discouraged over a basic close 
https://stackoverflow.com/questions/30889984/whats-the-difference-between-closeablehttpresponse-close-and-httppost-release
And you might be able to use a "try with resources" for closing

Reply:
weren't we trying to figure out the "default" httpclient we should use? IIRC, the last time I used an httpclient it wasn't apache in order to circumvent closing response and client

Reply:
Hmm, looks like you used the Java HTTP Client [here](https://github.com/dockstore/dockstore/blob/d34af609a98da26d6af2fdf57ae6f6d4362a96df/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java#L108). I'll try using that